### PR TITLE
fix: time unit bug/dockerfile/add a guide

### PR DESCRIPTION
**Pull request by chromevoid**
numaflow uses millisecond, while the python sdk code uses second.

fix the time unit inconsistency bug.
fix the dockerfile in flatmap example.
add a developer guide.. let me know if we want to put it elsewhere.

# Test
tested using the `developer_guide`. 
## Pipeline
```yaml
apiVersion: numaflow.numaproj.io/v1alpha1
kind: Pipeline
metadata:
  name: pynumasdk
spec:
  vertices:
    - name: in
      source:
        # A self data generating source
        generator:
          rpu: 10
          duration: 1s
    - name: cat
      udf:
        container:
          # compute the sum
          image: quay.io/numaio/numaflow-python/map-forward-message:latest
    - name: counter
      udf:
        container:
          # compute the sum
          image: quay.io/numaio/numaflow-python/reduce-counter:latest
        groupBy:
          window:
            fixed:
              length: 10s
          keyed: true
    - name: sink
      scale:
        min: 1
      sink:
        udsink:
          container:
            image: quay.io/numaio/numaflow-python/sink-log:latest
  edges:
    - from: in
      to: cat
    - from: cat
      to: counter
      parallelism: 1
    - from: counter
      to: sink
```
## Result
image
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ clean:
 	@find . -type f -name "*.py[co]" -exec rm -rf {} +
 
 format: clean
-	poetry run black pynumaflow/
+	poetry run black .
```
Comment (chromevoid) [inline Makefile:10]: Update it to `.` because `poetry run flake8 .` is applied to `.`

File: pynumaflow/function/server.py [hunk 2]
```diff
@@ -152,8 +152,8 @@ def ReduceFn(
                 f"got key: {key}, start: {start}, end: {end}."
             )
 
-        start_dt = datetime.fromtimestamp(int(start), timezone.utc)
-        end_dt = datetime.fromtimestamp(int(end), timezone.utc)
+        start_dt = datetime.fromtimestamp(int(start) / 1000, timezone.utc)
```
Comment (vigith) [inline pynumaflow/function/server.py:155]: @ab93 we cannot support milliseconds in python?

Comment (chromevoid) [reply pynumaflow/function/server.py:155]: It is milliseconds.

```sh
>>> print(datetime.fromtimestamp(1661169600000/1000, timezone.utc))
2022-08-22 12:00:00+00:00
>>> print(datetime.fromtimestamp(1661169600060/1000, timezone.utc))
2022-08-22 12:00:00.060000+00:00
```

Comment (ab93) [reply pynumaflow/function/server.py:155]: This is the preferred way from what I can understand. We don't lose granularity because it converts to float.

**Review by vigith — APPROVED** [review]

File: pynumaflow/function/server.py (hunk 2 above)
Comment (ab93) [reply pynumaflow/function/server.py:155]:
```suggestion
        start_dt = datetime.fromtimestamp(int(start) / 1e3, timezone.utc)
```

**Review by ab93 — COMMENTED** [review]
Can you also run `poetry version patch`?

**Review by ab93 — APPROVED** [review]

**Pull request merged**